### PR TITLE
[Convict] get autocomplete for format

### PR DESCRIPTION
**Pull request by rhapsodyn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/mozilla/node-convict#validation>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by msuntharesan — APPROVED** [review]

File: types/convict/index.d.ts [hunk 1]
```diff
@@ -37,6 +37,20 @@ declare namespace convict {
...
+        | 'port'
+        | 'windows_named_pipe'
+        | 'port_or_windows_named_pipe'
+        | 'url'
+        | 'email'
+        | 'ipaddress'
+        | 'duration'
+        | 'timestamp'
+        | 'nat'
+        | String;
```
Comment (amcasey) [inline types/convict/index.d.ts]: Did you mean `string`?

Comment (rhapsodyn) [reply types/convict/index.d.ts]: `string` cant get autocompletion in vscode, but `String` can (dont know why). Tested in vscode 1.29.1 MacOS.

Comment (amcasey) [reply types/convict/index.d.ts]: Sorry, I'm not sure what you mean by "get autocompletion".  What behavior is missing/incorrect for `string`?

Comment (vesse) [reply types/convict/index.d.ts]: These are used in convict like this:

```typescript
const config = {
  redis: {
    doc: 'Redis URL',
    format: String,
    default: 'redis://127.0.0.1:6379'
  }
}
```

and types cannot be used for the format - if you replace `String` with `string`, you'll see `'string' only refers to a type, but is being used as a value here.ts(2693)`.

**Review by vesse — CHANGES_REQUESTED** [review]

File: types/convict/index.d.ts [hunk 2]
```diff
@@ -37,6 +37,20 @@ declare namespace convict {
         parse: (content: string) => any;
     }
 
+    type PredefinedFormat =
```
[the remark this answers is not in the document]
Comment (amcasey) [reply types/convict/index.d.ts]: What happened to `Array`?

[the remark this answers is not in the document]
Comment (amcasey) [reply types/convict/index.d.ts]: I'm not sure that's equivalent.  What happens if you set `format` to `Array`?

**Review by vesse — APPROVED** [review]

File: types/convict/index.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (amcasey) [reply types/convict/index.d.ts]: @vesse, you're satisfied with the handling of `Array`?

[the remark this answers is not in the document]
Comment (rhapsodyn) [reply types/convict/index.d.ts]: Tested and no warning (using it my own code), with all ctors. @amcasey

[the remark this answers is not in the document]
Comment (amcasey) [reply types/convict/index.d.ts]: Great.  Thanks for testing!

**Pull request merged**